src/diffusers/training_utils.py

```diff
@@ -251,6 +251,32 @@ def state_dict(self) -> dict:
             "collected_params": self.collected_params,
         }
 
+    def store(self, parameters: Iterable[torch.nn.Parameter]) -> None:
+        r"""
+        Args:
+        Save the current parameters for restoring later.
+            parameters: Iterable of `torch.nn.Parameter`; the parameters to be
+                temporarily stored.
+        """
+        parameters = list(parameters)
+        self.collected_params = [param.detach().cpu().clone() for param in parameters]
+
+    def restore(self, parameters: Iterable[torch.nn.Parameter]) -> None:
+        r"""
+        Args:
+        Restore the parameters stored with the `store` method. Useful to validate the model with EMA parameters without:
+        affecting the original optimization process. Store the parameters before the `copy_to()` method. After
+        validation (or model saving), use this to restore the former parameters.
+            parameters: Iterable of `torch.nn.Parameter`; the parameters to be
+                updated with the stored parameters. If `None`, the parameters with which this
+                `ExponentialMovingAverage` was initialized will be used.
+        """
+        if self.collected_params is None:
+            raise RuntimeError("This ExponentialMovingAverage has no `store()`ed weights " "to `restore()`")
```

Comment:
If we do https://github.com/huggingface/diffusers/pull/2302/files#r1107495110, then this method wouldn't need to reference self and it wouldn't have this error condition. This would mean maybe it would look better as a function, and then store and restore could be free-standing functions not on a class.

WDYT?

Reply:
Since the `EMAModel` class majorly going to be used in the examples, which are meant to be easier to work with, I like this design approach better. So, with that in mind, for https://github.com/huggingface/diffusers/pull/2302#discussion_r1107495110, I think it's probably better if we do it ourselves. 

From what I have seen is that the community is fairly eager to customize things as per their needs, but with that, we should aim to provide example scripts that are easy as starting points. 

WDYT?

Reply:
I think we can still make the methods easy to work with without the adhoc variable storage. I.e.

```py
# training utils
def store(params):
    return [x.cpu().detach() for x in params]

def restore(params, stored_params):
    for c_param, param in zip(stored_params, parameters):
            param.data.copy_(c_param.data)


...

# training script
stored = store(params)
# do something
restore(params, stored)
```

is still friendly and hackable without the adhoc variable storage

Reply:
Then it is coming down to a preference now :D 

I like the class approach better than you showed above.